src/components/general/DatePicker.js

```diff
@@ -3,195 +3,132 @@ import PropTypes from "prop-types";
-				this.update({ day: day, month: item, year: year });
-		}
-		clickActionDay(item) {
-			const {
-				month,
-				year
-			} = this.state;
+		const daysInMonth = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
+		const monthInt = parseInt(month);
 
-			item = this.prepend0(item);
-			this.setState({ day: item });
+		// If the state's day is more than the max days in a month, set the day to the max day
+		const setMaxDay = (maxDay) => { if (day > maxDay) this.setState({ day: maxDay }); };
 
-			if (month !== "" && year !== "")
-				this.update({ day: item, month: month, year: year });
+		if (monthInt == 2 && this.isLeapYear(year)) {
+			setMaxDay(29);
+			return Array.from({ length: 29 }, (v, k) => k + 1);
 		}
+		setMaxDay(daysInMonth[monthInt - 1]);
+		return Array.from({ length: daysInMonth[monthInt - 1] }, (v, k) => k + 1);
+	}
 
-		clickActionYear(item) {
-			const {
-				month,
-				day
-			} = this.state;
+	/**
+	 * Returns true if the input year is a leap year. Must be an int.
+	 * @param {Number} year Any year such as 2020 or 101
+	*/
+	isLeapYear(year) { return (year % 4 === 0) && (year % 100 != 0) || (year % 400 == 0) }
 
-			this.setState({ year: item });
+	/** Sets the date by calling the onSelect function in the Form component. */
+	setDate(item) {
+		const { month, day, year } = item;
+		this.props.onSelect(`${year}-${month}-${day}`);
+	}
 
-			if (month === "2" && this.leapYear(item)) {
-				this.setState({ dayArr: Array.from({ length: 29 }, (v, k) => k + 1) });
-				if (day > 29) this.setState({ day: "29" });
-			}
-			if (month !== "" && day !== "")
-				this.update({ day: day, month: month, year: item });
+	/**
+	 * Calls the setDate function and sets the month/day/year based on a given value and type.
+	 * @param {Number} item Any number.
+	 * @param {String} type A string that is "month", "day", "year".
+	*/
+	setMDY(item, type) {
+		const { month, day, year } = this.state;
+		// Month and Day is prepended with a 0.
+		// Do note that the day January 1, 2000 will show up as 01/01/2000 instead of 1/1/2000.
+		switch (type) {
+			case "month": this.setState({ month: prepend0(item) });
+				break;
+			case "day": this.setState({ day: prepend0(item) });
+				break;
+			case "year": this.setState({ year: item });
+				break;
 		}
 
-	_keyExtractor = (item, index) => index;
+		if (month && day && year) this.setDate({ month, day, year });
+	}
 
 	render = () => {
-		const {
-			style,
-			datePickerStyle,
-			fieldContainer,
-			inputBoxStyle,
-			dropDownArrowStyle
-		} = styles;
-		const {
-			placeholder
-		} = this.props;
-		const {
-			month,
-			day,
-			year,
-			monthArr,
-			dayArr,
-			yearArr,
-			focused
-		} = this.state;
+		const { style, datePickerStyle, fieldContainer, inputBoxStyle, dropDownArrowStyle } = styles;
+		const { month, day, year, monthArray, yearArray } = this.state;
 
 		let iconSize = 32;
 
-		if (!focused)
-			return (
-				<View>
-					<Input
-						placeholder = { placeholder }
```

Comment:
Getting rid of this Input will not let the user know what information we want from them. They won't know if it's birthday, event day, or anything.